### PR TITLE
feat: Support PHPUnit 10

### PR DESCRIPTION
**Pull request by herndlm**
Makes it possible for consumers to use PHPUnit 10 by allowing `phpunit/php-text-template: ^3` (which only requires PHP 8.1+ from now on, see https://github.com/sebastianbergmann/php-text-template/blob/3.0.0/ChangeLog.md)

Additionally it allows PHPUnit 10 as a dev dependency and I fixed the reported warnings and deprecations, making sure it still works fine with e.g. PHP 7.4 and PHPUnit 9. I didn't go down further, I was assuming CI will be checking this if it's important, but the changes don't indicate that something was added that only a higher PHPUnit provides.

### CODE REVIEW

File: classes/functions/FixedDateFunction.php [hunk 1]
```diff
@@ -51,6 +51,6 @@ public function getCallable()
 
     public function increment($increment)
     {
-        $this->timestamp += $increment;
+        $this->timestamp += (int) $increment;
```
Comment (herndlm) [inline classes/functions/FixedDateFunction.php:54]: timestamp is an int and using e.g. float as $increment would implicitly cast it to int anyways, which was a deprecation warning.

File: tests/AbstractMockTestCase.php [hunk 2]
```diff
@@ -17,7 +17,7 @@
  * @license http://www.wtfpl.net/txt/copying/ WTFPL
  * @see Mock
  */
-abstract class AbstractMockTest extends TestCase
+abstract class AbstractMockTestCase extends TestCase
```
Comment (herndlm) [inline tests/AbstractMockTestCase.php:20]: PHPUnit 10 complained that the class is abstract, I figured that it of course can't run any tests from here if the class uses the `Test` prefix and removing / changing it makes it clear that there are no direct tests here

**Comment by herndlm** [comment]
saw the random CI git errors and force-pushed to trigger CI again, but unfortunately it still needs the approval because I didn't contribute yet

**Review by michalbundyra — APPROVED** [review]

**Comment by michalbundyra** [comment]
Thanks, @herndlm !

**Comment by herndlm** [comment]
thx too for doing the small needed adaptions to merge it :)

fyi my plan was to continue updating dependencies throughout packages to finally make mockery with php-mock/php-mock-mockery work with PHPUnit 10 as well over the next days

**Pull request merged**